x-pack/plugins/index_management/server/routes/api/inference_models/register_create_elasticsearch_inference_route.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { addBasePath } from '..';
+import { RouteDependencies } from '../../../types';
+
+export function registerCreateElasticsearchInferenceRoute({
+  router,
+  lib: { handleEsError },
+  config,
+}: RouteDependencies) {
+  const paramsSchema = schema.object({
+    inferenceId: schema.string(),
+  });
+  const bodySchema = schema.object({
+    trainedModelId: schema.string(),
+  });
+  // Create text_embedding inference
+  router.post(
+    {
+      path: addBasePath('/inference/text_embedding/{inferenceId}'),
+      validate: { body: bodySchema, params: paramsSchema },
+    },
+    async (context, request, response) => {
+      const { inferenceId } = request.params as typeof paramsSchema.type;
```

Comment:
You shouldn't need to do these `as typeof` typings, this should be typed correctly already. If it's not, we're doing something wrong.

Reply:
Updated